tests/serviceuser_test.go

```diff
@@ -119,11 +100,10 @@ func TestServiceUserKafka(t *testing.T) {
 	assert.Equal(t, map[string]string{"foo": "bar"}, secret.Annotations)
 	assert.Equal(t, map[string]string{"baz": "egg"}, secret.Labels)
 
-	// This kafka has sasl enabled and cert auth disabled.
-	// Which means that the port is not the same as in uri params.
 	strPort := string(secret.Data["SERVICEUSER_PORT"])
 	assert.NotEmpty(t, kafkaAvn.ServiceUriParams["port"])
-	assert.NotEqual(t, kafkaAvn.ServiceUriParams["port"], strPort)
+	// In case if Kafka has SASL enabled and cert auth disabled, the port would not be the same as in uri params
+	assert.Equal(t, kafkaAvn.ServiceUriParams["port"], strPort)
```

Comment:
I understand the comment, but it doesn't make sense now.
There was actually a real case for this https://github.com/aiven/aiven-operator/pull/565
I would let this test run its own Kafka.